src/main/js/components/TableView.jsx

```diff
@@ -1,40 +1,81 @@
-var React = require('react');
-var AmpersandMixin = require('ampersand-react-mixin');
-var Dispatcher = require('../dispatcher/Dispatcher');
-var Table = require('./table/Table.jsx');
-var LanguageSwitcher = require('./header/LanguageSwitcher.jsx');
-var TableSwitcher = require('./header/tableSwitcher/TableSwitcher.jsx');
-var ActionCreator = require('../actions/ActionCreator');
-var Tables = require('../models/Tables');
-var FilteredSubcollection = require('ampersand-filtered-subcollection');
-var RowConcatHelper = require('../helpers/RowConcatHelper');
+import React from "react";
+import connectToAmpersand from "../helpers/connectToAmpersand";
+import Dispatcher from "../dispatcher/Dispatcher";
+import Table from "./table/Table.jsx";
+import LanguageSwitcher from "./header/LanguageSwitcher.jsx";
+import TableSwitcher from "./header/tableSwitcher/TableSwitcher.jsx";
+import ActionCreator from "../actions/ActionCreator";
+import Tables from "../models/Tables";
 import * as AccessControl from "../helpers/accessManagementHelper";
 import * as _ from "lodash";
 import * as f from "lodash/fp";
-import TableauxConstants, {SortValues, ActionTypes, FilterModes} from "../constants/TableauxConstants";
+import TableauxConstants, {SortValues, ActionTypes, FilterModes, ColumnKinds} from "../constants/TableauxConstants";
 import Filter from "./header/filter/Filter.jsx";
 import Navigation from "./header/Navigation.jsx";
 import PageTitle from "./header/PageTitle.jsx";
 import Spinner from "./header/Spinner.jsx";
 import TableSettings from "./header/tableSettings/TableSettings";
-import searchFunctions from "../helpers/searchFunctions";
 import ColumnFilter from "./header/ColumnFilter";
 import {either} from "../helpers/monads";
+import {PAGE_SIZE, INITIAL_PAGE_SIZE} from "../models/Rows";
+import LocationBar from "location-bar";
+import getFilteredRows from "./table/RowFilters";
+import i18n from "i18next";
 
-var ColumnKinds = TableauxConstants.ColumnKinds;
+//hardcode all the stuffs!
+const ID_CELL_W = 80;
+const CELL_W = 300;
+const CELL_H = 46;
 
+@connectToAmpersand
 class TableView extends React.Component {
-//  mixins: [AmpersandMixin],
 
   constructor(props) {
     super(props);
     this.nextTableId = null;
+    this.pendingCellGoto = null;
+    this.tableFullyLoaded = false;
     this.state = {
       initialLoading: true,
       currentTableId: this.props.tableId,
       rowsCollection: null,
       rowsFilter: null
+    };
+
+    const {columnId, rowId, filter} = this.props;
+    if (columnId && rowId) {
+      this.pendingCellGoto = {
+        page: this.estimateCellPage(rowId),
+        row: rowId,
+        column: columnId,
+        filter: filter
+      }
     }
+
+    this.locationBar = new LocationBar();
```

Comment:
Do we really need `LocationBar` here? Is it possible to use `ampersand-router` for this?